### DISCUSSION TITLE
Any way to fire a hook after PR is created?

### DISCUSSION THREAD

**Discussion by RyuunosukeDS3** [renovatebot/renovate#23037] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.98.4

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to be able to execute a custom script after every MR(PR) renovate creates. We are currently integrating renovate in our workflow. And it's been great, but it is important that we are able to collect info on the MRs created and send it to a database  then analyse it on grafana in order to know how our teams are dealing with it, so we can better make an strategy before we roll out renovate as one of our tools in our workflow.

### What you tried so far.

I've tried looking in the documentation but only saw pre-commit...

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
The closest you can get is https://docs.renovatebot.com/configuration-options/#postupgradetasks, which runs after updates are done but before commit/PR

**Reply by RyuunosukeDS3** [reply]
Any chance we can get this feature in the foreseeeble future? Or is it just too out of the scope?

**Reply by rarkins** [reply]
If you're willing to implement it, then we can help you design the feature

**Reply by rarkins** [reply]
Alternatively if your company is willing to sponsor the development, there are developers who take on freelance work implementing features

**Reply by RyuunosukeDS3** [reply]
I might implement it. Maybe this weekend. I will take a look at the code and see how much effort I will have to put, if it is reasonable I will implement it myself. Thanks for the answers!

**Reply by rarkins** [reply, marked as answer]
I have created a feature request issue in #23050
